types/example.pp

```diff
@@ -1,14 +1,14 @@
 type Patterndb::Example = Struct[
   {
-    program      => String[1],
+    program      => String,
     test_message => String[1],
     test_values  => Optional[
       Hash[
         String[1],
         Struct[
           {
-            type  => Optional[Enum['datetime', 'double', 'integer', 'json', 'string']],
-            value => String[1],
+            type  => Optional[Enum['datetime', 'double', 'integer', 'json', 'string', 'null']],
+            value => String,
```

Comment:
```suggestion
            value => String[0],
```

Reply:
I agree it's more concise, but both forms are functionally [identical](https://www.puppet.com/docs/puppet/7/lang_data_string#lang_data_string_data_type)

Reply:
I think at some point I will hack a puppet-lint plugin to find these in the puppetlabs / voxpupuli modules.  The fact that (most) String already have a lower bound here when required mean that it will be easy for us to fix those if needed in the future.